tests/bug-reports/Dec.fst

```diff
@@ -0,0 +1,107 @@
+module Dec
+
+let rec f (x y:nat)
+  : Tot unit (decreases y)
+  = if y = 0 then ()
+    else f (x + 1) (y - 1)
+
+let rec f1 (x y :nat)
+  : Tot unit (decreases y)
+  = if y = 0 then ()
+    else f2 (x + 1) (y - 1)
+
+and f2 (x y:nat)
+  : Tot unit (decreases y)
+  = if y = 0 then ()
+    else f1 (x + 1) (y - 1)
+
+let pred (x y:nat) : prop = (x <= y) == true
+let rec g (x' y' : nat)
+  : Lemma (requires y' >= x')
+          (ensures (pred x' y'))
+          (decreases y')
+  = if x' > 0
+    then g (x' - 1) (y' - 1)
+    else ()
+
+
+[@@expect_failure [19]]
```

Comment:
This is testing that the precondition of the `val` implies that of the `let`?

Reply:
yep